src/main/java/com/jfrog/ide/common/utils/Utils.java

```diff
@@ -127,4 +128,42 @@ public static Set<Path> consolidatePaths(Set<Path> paths) {
     public static String removeComponentIdPrefix(String compId) {
         return StringUtils.substringAfter(compId, "://");
     }
+
+    /**
+     * Resolve Xray URL from the input Xray URL (may be blank) or from the input platform URL.
+     *
+     * @param xrayUrl     - Customize Xray URL or blank
+     * @param platformUrl - JFrog platform URL
+     * @return the resolved Xray URL
+     */
+    public static String resolveXrayUrl(String xrayUrl, String platformUrl) {
+        String url = trimToEmpty(xrayUrl);
+        if (isNotBlank(url)) {
+            return url;
+        }
+        url = trimToEmpty(platformUrl);
+        if (isBlank(url)) {
+            return "";
+        }
+        return removeEnd(url, "/") + "/xray";
+    }
+
+    /**
+     * Resolve Artifactory URL from the input Artifactory URL (may be blank) or from the input platform URL.
+     *
+     * @param artifactoryUrl - Customize Artifactory URL or blank
+     * @param platformUrl    - JFrog platform URL
+     * @return the resolved Xray URL
+     */
+    public static String resolveArtifactoryUrl(String artifactoryUrl, String platformUrl) {
```

Comment:
Most of this implementation is identical to `resolveXrayUrl` implementation. Can we move it to a shared private function?

Reply:
Good point. Thanks!